src/providers/atcoder.cr

```diff
@@ -0,0 +1,31 @@
+class AtCoderStats < SiteStats
+  @@url_regex = Regex.new("https?://(www[.])?atcoder.jp/users/(?<username>\\w+)")
```

Comment:
the `.` in `.jp` should be escaped, also does atcoder allow usernames that have non-word characters like `-` or `.`?